core/dbt/parser/read_files.py

```diff
@@ -3,7 +3,8 @@
     FilePath, ParseFileType, SourceFile, FileHash, AnySourceFile, SchemaSourceFile
 )
 
-from dbt.parser.schemas import yaml_from_file
+from dbt.parser.schemas import yaml_from_file, schema_file_keys, check_format_version
+from dbt.exceptions import CompilationException
```

Comment:
For later, not now: We may want a cleaner distinction between compile-time exceptions (Jinja rendering, SQL templating) and validation/parse-time exceptions. This would be one of the latter.